src/com/owncloud/android/ui/activity/ManageSpaceActivity.java

```diff
@@ -0,0 +1,130 @@
+package com.owncloud.android.ui.activity;
+
+import android.app.Activity;
+import android.app.ActivityManager;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.owncloud.android.R;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+import java.io.File;
+
+public class ManageSpaceActivity extends AppCompatActivity {
+
+    private static final String TAG = ManageSpaceActivity.class.getSimpleName();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_manage_space);
+
+        ActionBar actionBar = getSupportActionBar();
+        actionBar.setDisplayHomeAsUpEnabled(true);
+        actionBar.setTitle(R.string.manage_space_title);
+
+        TextView descriptionTextView = (TextView) findViewById(R.id.general_description);
+        descriptionTextView.setText(getString(R.string.manage_space_description, getString(R.string.app_name)));
+
+        Button clearDataButton = (Button) findViewById(R.id.clearDataButton);
+        clearDataButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                clearData();
+            }
+        });
+    }
+
+    /**
+     * Save passcode from Share preferences
+     * Clear the rest of data
+     */
+    private void clearData() {
+        // Save passcode from Share preferences
+        SharedPreferences appPrefs = PreferenceManager
+                .getDefaultSharedPreferences(getApplicationContext());
+
+        boolean passCodeEnable = appPrefs.getBoolean("set_pincode", false);
+
+        String passCodeDigits[] = new String[4];
+        if (passCodeEnable) {
+            passCodeDigits[0] = appPrefs.getString("PrefPinCode1", null);
```

Comment:
Now "PrefPinCode*" are static final string in PassCodeActivity.

IMHO, the passcode storage scheme must be changed in other issue, because that change needs different tests than #1325.